erpnext/erpnext_integrations/doctype/plaid_settings/plaid_settings.py

```diff
@@ -2,37 +2,43 @@
 # Copyright (c) 2018, Frappe Technologies Pvt. Ltd. and contributors
 # For license information, please see license.txt
 
-from __future__ import unicode_literals
-import frappe
 import json
-from frappe import _
-from frappe.model.document import Document
+
+import frappe
 from erpnext.accounts.doctype.journal_entry.journal_entry import get_default_bank_cash_account
 from erpnext.erpnext_integrations.doctype.plaid_settings.plaid_connector import PlaidConnector
-from frappe.utils import getdate, formatdate, today, add_months
+from frappe import _
 from frappe.desk.doctype.tag.tag import add_tag
+from frappe.model.document import Document
+from frappe.utils import add_months, formatdate, getdate, today
+
 
 class PlaidSettings(Document):
-	pass
+	def get_link_token(self):
```

Comment:
When you're creating an object from PlaidConnector, you're triggering the `__init__` method (https://github.com/ParsimonyGit/erpnext/pull/3/files#diff-4a2f5267f7afd109d18bfb6442a6e552R17) in that class which is in turn creating an object of this (Plaid Settings) class...

Dependency loop-de-loop aside, you're triggering a query twice to get the same data.

Reply:
Hmm, I'll change this into a staticmethod then. I don't really need the class instance for the token.